src/Build.UnitTests/BackEnd/TaskRegistry_Tests.cs

```diff
@@ -622,6 +622,99 @@ public void RetrieveFromCacheMatchingExactParameters_AdditionalParameters()
                 );
         }
 
+        [Fact]
+        public void OverriddenTask_AlwaysWins()
+        {
```

Comment:
Can you combine these two Facts into a Theory then add a case for element having Architecture = "x64" and Override set to "false" then secondElement having Architecture "x86" and Override "true"?